templates/files/conf/attach-etcd-disk

```diff
@@ -0,0 +1,152 @@
+WAA_FILE="/var/lib/waagent/provisioned"
+until test -f ${WAA_FILE}; do
+    echo "Waiting for waagent.service to finish provisioning."
+    sleep 5
+done
+sleep 30s
+
+LUN=20
+METADATA_ENDPOINT="http://169.254.169.254/metadata/instance?api-version=2020-09-01"
+
+DISK=/dev/disk/azure/scsi1/lun${LUN}
+MOUNTPOINT=/var/lib/etcd
+
+metadata="$(curl -H Metadata:true --noproxy "*" "$METADATA_ENDPOINT" -s)"
+
+SUBSCRIPTION="$(echo "$metadata" | jq -r .compute.subscriptionId)"
+RESOURCEGROUP="$(echo "$metadata" | jq -r .compute.resourceGroupName)"
+VMSS="$(echo "$metadata" | jq -r .compute.vmScaleSetName)"
+resourceid="$(echo "$metadata" | jq -r .compute.resourceId)"
+# Extract instance ID (a number) from the resource ID.
+INSTANCEID="$(echo $resourceid | grep -Eo '[0-9]+$')"
+
+run_az_cmd () {
+  cmd="$1"
+
+  docker run --rm mcr.microsoft.com/azure-cli bash -c "az login --identity 2>&1 >/dev/null && $cmd"
+}
+
+if [ -f "$DISK" ] || [ -h "$DISK" ]
+then
+  echo "There is a disk attached on lun ${LUN}"
+  echo "Checking disk attached to lun ${LUN}"
+
+  attached_disk_id="$(echo "$metadata" | jq -r --arg lun "$LUN" '.compute.storageProfile.dataDisks[] | select(.lun == $lun).managedDisk.id')"
+  echo "Disk ID is $attached_disk_id"
+
+  disk_details="$(run_az_cmd "az disk show --ids $attached_disk_id")"
+
+  role="$(echo "$disk_details" | jq -r .tags.GiantSwarmRole)"
+
+  echo "role is $role"
+
+  if [ "$role" != "etcd" ]
+  then
+    echo "Expected disk $attached_disk_id to have a tag GiantSwarmRole with value 'etcd' but that wasn't the case. Aborting."
+    exit 7
+  fi
+  MASTER_ID="$(echo "$disk_details" | jq -r .tags.GiantSwarmEtcdID)"
+  echo "MASTER_ID is $MASTER_ID"
+  if [ "$MASTER_ID" == "null" ]
+  then
+    echo "Disk $attached_disk_id is missing the GiantSwarmEtcdID tag, can't continue"
+    exit 8
+  fi
+
+  export MASTER_ID
+else
+  # list all disks with tag GiantSwarmRole == "etcd" and having a GiantSwarmEtcdID tag
+  echo "Looking for available disks"
+  # TODO discard disks with multi-attach feature enabled when az will support it
+  candidates="$(run_az_cmd "az disk list -g ${RESOURCEGROUP} | jq '. | map(select(.tags.GiantSwarmRole == \"etcd\" and .diskState == \"Unattached\" and (.tags | has(\"GiantSwarmEtcdID\"))))'")"
+
+  if [ "$candidates" == "[]" ]
+  then
+    echo "No unattached disks with a tag 'GiantSwarmRole' = 'etcd' and a 'GiantSwarmEtcdID' tag were found, can't continue"
+    exit 5
+  fi
+
+  # try attaching one disk from the list randomly in a loop until it succeedes
+  names="$(echo "$candidates" | jq -r .[].name)"
+  echo "Found some disks to try attaching: $names"
+
+  for diskname in $names
+  do
+    echo "Trying to attach disk $diskname to lun $LUN"
+    set +e
+    diskid="/subscriptions/${SUBSCRIPTION}/resourcegroups/${RESOURCEGROUP}/providers/Microsoft.Compute/disks/${diskname}"
+    # TODO un-hardcode registry
+    IMAGE="quay.io/giantswarm/azure-disk-attacher:latest"
```

Comment:
Can we configure the docker mirroring here? If not, maybe we can use ACR instead of quay here, since this is just for azure? https://portal.azure.com/#blade/Microsoft_Azure_ContainerRegistries/TagMetadataBlade/id/%2Fsubscriptions%2F1be3b2e6-497b-45b9-915f-eb35cae23c6a%2FresourceGroups%2Fgiantswarm-container-registry%2Fproviders%2FMicrosoft.ContainerRegistry%2Fregistries%2Fgiantswarm/repository/giantswarm%2Fazure-disk-attacher

Reply:
good catch, let me check how we deal with images in MCs

Reply:
According to biscuit we don't have registry mirroring in MCs, so this is not making nodes any less reliable than they already are. Does it work for you @fiunchinho ?

Reply:
I'm fine merging it as it is. But if we don't have mirroring, is there a reason NOT to use the azure registry, given that the image is already there?

Reply:
I guess the reason is consistency to the other units that use quay.io?

Reply:
I'd argue reliability > consistency.

Anyway, your call.